ebos/eclproblem.hh

```diff
@@ -1459,103 +1459,105 @@ public:
-        assert(bcindex_(dir)[globalDofIdx] > 0);
-        const auto& bc = this->simulator().vanguard().schedule()[this->episodeIndex()].bcprop[bcindex_(dir)[globalDofIdx]];
-        if (bc.bctype == BCType::DIRICHLET )
-        {
-            InitialFluidState fluidState;
-            const int pvtRegionIdx = this->pvtRegionIndex(globalDofIdx);
-            fluidState.setPvtRegionIndex(pvtRegionIdx);
-
-            switch (bc.component) {
-                case BCComponent::OIL:
-                    if (!FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx))
-                        throw std::logic_error("oil is not active and you're trying to add oil BC");
-
-                    fluidState.setSaturation(FluidSystem::oilPhaseIdx, 1.0);
-                    break;
-                case BCComponent::GAS:
-                    if (!FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx))
-                        throw std::logic_error("gas is not active and you're trying to add gas BC");
-
-                    fluidState.setSaturation(FluidSystem::gasPhaseIdx, 1.0);
-                    break;
-                    case BCComponent::WATER:
-                    if (!FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx))
-                        throw std::logic_error("water is not active and you're trying to add water BC");
-
-                    fluidState.setSaturation(FluidSystem::waterPhaseIdx, 1.0);
-                    break;
-                case BCComponent::SOLVENT:
-                case BCComponent::POLYMER:
-                case BCComponent::NONE:
-                    throw std::logic_error("you need to specify a valid component (OIL, WATER or GAS) when DIRICHLET type is set in BC");
-                    break;
-            }
-            int phaseIndex;
-            if (FluidSystem::phaseIsActive(oilPhaseIdx)) {
-                phaseIndex = oilPhaseIdx;
-            }
-            else if (FluidSystem::phaseIsActive(gasPhaseIdx)) {
-                phaseIndex = gasPhaseIdx;
-            }
-            else {
-                phaseIndex = waterPhaseIdx;
-            }
-            double pressure = initialFluidStates_[globalDofIdx].pressure(phaseIndex);
-            const auto pressure_input = bc.pressure;
-            if (pressure_input) {
-                pressure = *pressure_input;
-            }
-
-            std::array<Scalar, numPhases> pc = {0};
-            const auto& matParams = materialLawParams(globalDofIdx);
-            MaterialLaw::capillaryPressures(pc, matParams, fluidState);
-            Valgrind::CheckDefined(pressure);
-            Valgrind::CheckDefined(pc);
-            for (unsigned phaseIdx = 0; phaseIdx < numPhases; ++phaseIdx) {
-                if (!FluidSystem::phaseIsActive(phaseIdx))
-                    continue;
+        const auto& bcprop = this->simulator().vanguard().schedule()[this->episodeIndex()].bcprop;
+        if (bcprop.size() > 0){
```

Comment:
If I understand correctly, unless we have a dirichlet condition we just return the initialFluidStates_[globalDofIdx], and that is not changed by this PR. The new if her is just to avoid indexing into an empty bcprop?

Nitpick: space before {